### PR TITLE
Simulation

### PR DESCRIPTION
**Pull request by irapha**
PR not ready. Here for tracking changes

### CODE REVIEW

File: buzzmobile/drive/car_interface/src/main.cpp [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #include <ros/ros.h>
 #include "arduino.h"
-//#include <core_msgs/Odom.h>
 #include <buzzmobile/CarPose.h>
-//#include <core_msgs/MotionCommand.h>
 #include <std_msgs/Bool.h>
+//#include <core_msgs/Odom.h>
+//#include <core_msgs/MotionCommand.h>
```
Comment (irapha) [inline buzzmobile/drive/car_interface/src/main.cpp]: We're not using these msgs, can you remove these includes?

File: buzzmobile/launch/simulation.launch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <group ns="buzzmobile">
+        <!-- include file="$(find buzzmobile)/launch/includes/params.launch"/ -->
+        <!-- include file="$(find buzzmobile)/launch/includes/gps.launch"/ -->
+        <!-- include file="$(find buzzmobile)/launch/includes/mapping.launch"/ -->
+        <!-- include file="$(find buzzmobile)/launch/includes/steering.launch"/ -->
+        <!-- include file="$(find buzzmobile)/launch/includes/lidar.launch"/ -->
+        <!-- include file="$(find buzzmobile)/launch/includes/controller.launch"/ -->
+        <!-- group if="$(arg mission_control)" -->
+            <!-- include file="$(find buzzmobile)/launch/includes/visualization.launch"/ -->
+        <!-- /group -->
```
Comment (irapha) [inline buzzmobile/launch/simulation.launch]: can you uncomment all of these? this file should load up bzm too.

File: buzzmobile/launch/simulation.launch [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <!-- Start gazebo server, load world, optionally start viz client -->
+    <node pkg="gazebo_ros" name="gazebo" type="gzserver"
+          args="-e ode $(arg world) use_sim_time:=true"
+          output="screen"/>
+    <group if="$(arg gzclient)">
+        <node pkg="gazebo_ros" name="gazebo_gui" type="gzclient"/>
+    </group>
+
+    <!-- Spawn car -->
+    <!-- node pkg="buzzmobile" name="sim_spawn_car" type="sim_spawn_car.sh"/ -->
```
Comment (irapha) [inline buzzmobile/launch/simulation.launch:28]: since this is working without the bash script, can we delete that and close #158 ?

Comment (irapha) [reply buzzmobile/launch/simulation.launch:28]: Or do you think it's worth waiting until we're loading up a real world (not `empty.world`) to test if this still works?

File: buzzmobile/launch/test.launch [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+   <node pkg="controller_manager" name="spawner" type="spawner"
+         args="left_wheel_effort_controller right_wheel_effort_controller \
+         left_wheel_position_controller right_wheel_position_controller"/>
+
+    <!-- Load Joint Mappings and Service Controllers -->
+    <rosparam command="load" file="$(find buzzmobile)/simulation/models/car/service_controllers_to_joints_mapping.yaml"/>
+
+    <!-- Spawn Car -->
+    <param name="robot_description" command="cat $(find buzzmobile)/simulation/models/car/car_joints.urdf"/>
```
Comment (irapha) [inline buzzmobile/launch/test.launch]: can you delete this file?

File: buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+void carPoseCallback(const buzzmobile::CarPose::ConstPtr &msg) {
+    speed_set_point = -msg->velocity;
+    steer_set_point = -msg->angle;
+
+    // arduino.setSteering(cmd->angle);
+    // arduino.setHorn(cmd->horn);
+    // last_command_time = cmd->header.stamp;
+}
+
+// void steeringCallback(const rr_platform::steeringConstPtr &msg) {
```
Comment (irapha) [inline buzzmobile/simulation/sim_car_interface/src/sim_car_interface.cpp]: can you remove all the commented out things in this file?

**Review by irapha — COMMENTED** [review]
Oops, turns out I made this PR. Can anyone make these changes and I'll review the PR again?

File: buzzmobile/launch/simulation.launch (hunk 3 above)
Comment (irapha) [reply buzzmobile/launch/simulation.launch:28]: I'm gonna wait until we have real worlds to test that this still works

**Review by irapha — COMMENTED** [review]
LGTM, but I can't approve this PR because I'm the one who started it. @joshuamorton can you please?

**Comment by joshuamorton** [comment]
Sure, one question before we do: can we get a (super barebones) `RosTest` that works using this branch and multiple-roscore.

nothing that needs to be merged in, but a sanity check that they will interact well.

File: install [hunk 6]
```diff
@@ -96,3 +96,4 @@ echo "alias rosdevel='source ~/catkin_ws/devel/setup.bash' # needed for autocomp
 echo "alias rosvenv='source ~/catkin_ws/src/buzzmobile/bin/activate' # activate venv" >> ~/.bashrc
 echo "rosinit     # put these in your .bashrc too so they run everytime" >> ~/.bashrc
 echo "rosdevel    # you open a new terminal window/tab." >> ~/.bashrc
+echo "export SVGA_VGPU10=0" >> ~/.bashrc # disables support for OpenGL 3.3, allows gzclient to run
```
Comment (joshuamorton) [inline install:99]: Uhhh, shouldn't this be done as part of rosinit, and not just done all the time? or will this not cause unwanted side effects?

**Review by joshuamorton — COMMENTED** [review]
See that one comment, also could we potentially convert sim_car_interface to python? IDK if that's worth or not.

File: install (hunk 6 above)
Comment (irapha) [reply install:99]: rosinit is done all the time anyways, but nevertheless, I don't think it will cause unwanted side effects.

**Comment by irapha** [comment]
About the test, I can write something that will literally spin up the launch file, and assert true.

I can't guarantee that we'll be able to write a test that will actually check if actuators work though, because of #161 

Also I didn't want to merge #180 into this PR

File: install (hunk 6 above)
Comment (joshuamorton) [reply install:99]: (Y)

**Comment by joshuamorton** [comment]
Not merging 180 in is fine. As long as it passed locally on some branch that's fine. I didn't even necesarilly want the test written on here, just confirmation that it works.

**Review by joshuamorton — APPROVED** [review]

**Comment by irapha** [comment]
Oh Alright. Just tested locally in a new merged branch and it works. I'll revert the file in this PR and we can merge.

Make a note somewhere in a readme for test utils that when issues arise its a good idea to clean pyc files. I was having trouble with a test hanging (because a message wasnt being received so it just waited on stdin input forever) and cleaning pyc files fixed it.

**Comment by joshuamorton** [comment]
noting that this closes #182, #160, and #159

**Pull request merged**